<launch>

    <arg name="markerId"        default="144"/>
    <arg name="markerSize"      default="0.048"/>    <!-- in m -->
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="SUBPIX" /> <!-- NONE, HARRIS, LINES, SUBPIX -->


    <node pkg="aruco_ros" type="single" name="aruco_single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_rect_color" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="camera_color_optical_frame"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

	<node name="vrep_combiner" pkg="util_pkg" type="combiner" args="" respawn="false" output="screen">>
		
		<!-- if false, it work on coppeliasim with rosInterface plugin-->
		<param name="old_plugin" type="bool" value="false"/> 
		
		<!-- subscribers topic-->
		<param name="camera_color_name" type="string" value="/camera/color/image_rect_color"/>
		<param name="camera_depth_name" type="string" value="/camera/depth/image_rect_color"/>
		<param name="pcd_topic" type="string" value="/camera/depth/image"/>
	</node>


	<node name="rviz_peg_ring" pkg="rviz" type="rviz"
	    args="-d $(find util_pkg)/rviz_scenes/aruco_global_frame.rviz"
	    respawn="false"/> 

</launch>